core/modules/order/hooks/beforeRegistration.ts

```diff
@@ -1,10 +1,122 @@
 import * as localForage from 'localforage'
 import UniversalStorage from '@vue-storefront/core/store/lib/storage'
+import EventBus from '@vue-storefront/core/compatibility/plugins/event-bus/index'
+import { Logger } from '@vue-storefront/core/lib/logger'
+import { currentStoreView } from '@vue-storefront/core/lib/multistore';
+import rootStore from '@vue-storefront/core/store'
+import i18n from '@vue-storefront/i18n'
+import { serial, onlineHelper } from '@vue-storefront/core/helpers'
 
 export function beforeRegistration ({ Vue, config, store, isServer }) {
   Vue.prototype.$db.ordersCollection = new UniversalStorage(localForage.createInstance({
     name: 'shop',
     storeName: 'orders',
     driver: localForage[config.localForage.defaultDrivers['orders']]
   }))
+  if (!isServer) {
+    const orderMutex = {}
+    // TODO: move to external file
+    EventBus.$on('order/PROCESS_QUEUE', async event => {
```

Comment:
Same as above, we should avoid using event bus whenever it's possible

Reply:
Let's have an issue for this